core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -625,11 +631,13 @@ public void testSerialization() throws IOException {
      * Serialize the given query builder and asserts that both are equal
      */
     @SuppressWarnings("unchecked")
-    protected QB assertSerialization(QB testQuery) throws IOException {
+    protected <QB extends QueryBuilder> QB assertSerialization(QB testQuery) throws IOException {
```

Comment:
since the QB type that is passed in can be different to the QB type of the class